addons/mail/models/mail_scheduled_message.py

```diff
@@ -0,0 +1,255 @@
+
+        allowed_ids = defaultdict(set)
+        for model, res_ids in model_ids.items():
+            records = self.env[model].browse(res_ids)
+            operation = records._mail_post_access
+            if records.has_access(operation):
+                allowed_ids[model] = set(records._filtered_access(operation)._ids)
+
+        scheduled_messages = self.browse(
+            id_
+            for id_, model, res_id in rows
+            if res_id in allowed_ids[model]
+        )
+
+        return scheduled_messages._as_query(order)
+
+    def unlink(self):
+        self._check()
+        return super().unlink()
+
+    def write(self, vals):
+        # prevent changing the records on which the messages are scheduled
+        if vals.get('model') or vals.get('res_id'):
+            raise UserError(_('You are not allowed to change the target record of a scheduled message.'))
+        # make sure user can write on the record the messages are scheduled on
+        self._check()
+        res = super().write(vals)
+        if new_scheduled_date := vals.get('scheduled_date'):
+            self.env.ref('mail.ir_cron_send_scheduled_message')._trigger(fields.Datetime.to_datetime(new_scheduled_date))
+        return res
+
+    # ------------------------------------------------------
+    # Actions
+    # ------------------------------------------------------
+
+    def open_edit_form(self):
+        self.ensure_one()
+        return {
+            'type': 'ir.actions.act_window',
+            'name': _("Edit Scheduled Note") if self.is_note else _("Edit Scheduled Message"),
+            'res_model': self._name,
+            'view_mode': 'form',
+            'views': [[False, 'form']],
+            'target': 'new',
+            'res_id': self.id,
+        }
+
+    def send(self):
+        self.ensure_one()
+        if self.create_uid.id != self.env.uid:
+            if self.env.is_admin():
+                self.with_user(self.create_uid)._send()
+            else:
+                raise UserError(_('You are not allowed to send this scheduled message'))
+        self._send()
+
+    def _send(self, raise_exception=True):
+        """ Send the scheduled messages
+            if raise_exception is set to False, the method will skip the posting of a message
+            instead of raising an error. This is useful when scheduled messages are sent from
+            the _send_messages_cron
+        """
+        messages = self.env['mail.message']
+        notification_parameters_whitelist = self._notification_parameters_whitelist()
+        for scheduled_message in self:
+            if raise_exception:
+                scheduled_message._check()
+            else:
+                try:
+                    scheduled_message._check()
+                except AccessError:
+                    _logger.info("Posting of scheduled message %s failed: user %s cannot post on the record", scheduled_message.id, scheduled_message.create_uid)
+                    continue
```

Comment:
Isn't it equivalent to
```suggestion
            try:
                scheduled_message._check()
            except AccessError:
                if raise_exception:
                    raise
                _logger.info("Posting of scheduled message %s failed: user %s cannot post on the record", scheduled_message.id, scheduled_message.create_uid)
                continue
```